### DISCUSSION TITLE
Is it possible to trigger a filter programmatically in tables?

### DISCUSSION THREAD

**Discussion by ekandreas** [filamentphp/filament#11176] [General]
I would like to trigger a filter to a table in a backend function. Is it possible?

**Comment by ekandreas** [comment]
Solved it in this way:
1. Add a public bool in the resource, eg: `public bool $favoritesFilter = false;`
2. Set the filters "default" to this bool, eg: `->default($this->favoritesFilter)`
3. From the function, toggle the prop and refresh the table, eg:
```
        $this->favoritesFilter = !$this->favoritesFilter;
        $this->resetTable();
```

**Comment by VictorPennington** [comment]
Hello there, I think I've found an alternative:

Apparently one can can append filters to the session and then reset the page, to apply the new filters.
This approach seems quite handy, specially because it doesn't destroy the other applied filters, if you have them.

Using table utility injections, you can access the [livewire ](https://filamentphp.com/docs/3.x/tables/filters/getting-started#injecting-the-current-livewire-component-instance)component, including filters, and can do that in a header Action, for example:

```

<?php

use Filament\Tables\Table;
use Filament\Tables\Contracts\HasTable;

public function table(Table $table): table
{ 
    return $table
        ->(...)
        ->headerActions([
            Action::make('today')
                ->action(function(HasTable $livewire) {
                    // Here you change the filter to your liking
                    $livewire->tableFilters['today']['isActive'] = !$livewire->tableFilters['today']['isActive'];

                    // Here you append to the session all the filters and reload.
                    session()->put(
                        $this->getTableFiltersSessionKey(),
                        $this->tableFilters
                    );
                    $this->resetPage();
                })
        ]);
}
````

**Reply by alekmlynek** [reply]
Thank you! This is exactly what I was looking for. 

I also wanted to comment on that, adding:

`$livewire->tableFilters['today']['isActive'] = !$livewire->tableFilters['today']['isActive'];`

It was enough for me. I did not need the rest. However, I had to place this in the ->after() function instead of the action().
Also, my IDE did not pick up $livewire->tableFilters[], but if you use it, it should work anyway.

**Comment by peteleco** [comment]
Thanks! You can also access this two methods: resetTableFiltersForm and updatedTableFilters to keep shorter
```php
        $livewire->resetTableFiltersForm(); // If you want to reset the filter before apply
        $livewire->tableFilters['today']['isActive'] = !$livewire->tableFilters['today']['isActive'];
        $livewire->updatedTableFilters();
```
